plugins/portal-history-ornaments.js

```diff
@@ -0,0 +1,244 @@
+      }).addTo(portal._historyLayer);
+    }
+    if (drawMissing && portal.options.data.history.visited && !portal.options.data.history.captured 
+        || !drawMissing && portal.options.data.history.visited && !portal.options.data.history.captured) {
+      L.marker(portal._latlng, {
+        icon: portalsHistory.iconSemiMarked[portal.options.level],
+        interactive: false,
+        keyboard: false,
+      }).addTo(portal._historyLayer);
+    }
+    if (drawMissing && !portal.options.data.history.scoutControlled || !drawMissing && portal.options.data.history.scoutControlled) {
+      L.marker(portal._latlng, {
+        icon: portalsHistory.iconScouted[portal.options.level],
+        interactive: false,
+        keyboard: false,
+      }).addTo(portal._historyLayer);
+    }
+  }
+  portal._historyLayer.addTo(portalsHistory.layerGroup);
+}
+
+function drawAllFlags () {
+  portalsHistory.layerGroup.clearLayers();
+  createIcons();
+  //IITC.me support: getCurrentZoomTileParameters is iitc.app only; iitc.me function is: getMapZoomTileParameters
+  var tileParams = window.getCurrentZoomTileParameters ? window.getCurrentZoomTileParameters() : window.getMapZoomTileParameters();
+  if (tileParams.level !== 0) {
+    return;
+  }
+
+  for (var id in window.portals) {
+    drawPortalFlags(window.portals[id]);
+  }
+}
+
+function getSVGString (size, color, parts, offset) {
+  var circumference = size * Math.PI;
+  var arcOffset = circumference / parts * (parts - 1);
+  var rotate = 360 / parts * offset;
+  return `<svg width="${(size+4)}" height="${(size+4)}" xmlns="http://www.w3.org/2000/svg">
+          <circle stroke="${color}" stroke-width="4" fill="transparent" cx="${(size+4)/2}" cy="${(size+4)/2}" 
+          r="${(size/2)}" stroke-dasharray="${circumference}" stroke-dashoffset="${arcOffset}" 
+          transform="rotate(${rotate}, ${((size+4)/2)}, ${((size+4)/2)})" />
+          </svg>`;
+}
+// -----------------------------------------------------------------------------------------
+var setup = function () {
+
+  var checkedCircle = '<svg class="tracker-eye" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"> <path d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"/></svg>';
+  var emptyCircle =  '<svg class="tracker-eye" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"> <path d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"/></svg>';
+
+  var style = `<style>
```

Comment:
Backticks

Reply:
I would not change this here atm, as we want to move the code to related .css we only need to do a Cut'n'Paste
What is the preferred way for the SVGs referenced in the styles?
Leave them in plugin and reference in CSS?
Is there a better way, that would not mean to split the code to two files?

Reply:
revamped the code.
SVG symbols added to body for future reuse (this is also a PoC for #236)